pkg/readiness/ready_tracker_unit_test.go

```diff
@@ -162,3 +214,60 @@ func Test_ReadyTracker_TryCancelTemplate_Retries(t *testing.T) {
 		t.Fatal("tracker with 0 retries and cancellation should be satisfied")
 	}
 }
+
+func Test_Tracker_TryCancelData(t *testing.T) {
+	l := lister{syncsets: true}
+	for _, tt := range []struct {
+		name      string
+		objDataFn func() objData
+	}{
+		{name: "no retries"},
+		{name: "with retries", objDataFn: func() objData {
+			return objData{retries: 2}
+		}},
+	} {
+		rt := newTracker(l, false, false, false, tt.objDataFn)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		var runErr error
+		runWg := sync.WaitGroup{}
+		runWg.Add(1)
+		go func() {
+			runErr = rt.Run(ctx)
+			runWg.Done()
+		}()
+
+		require.Eventually(t, func() bool {
+			return rt.Populated()
+		}, 10*time.Second, 1*time.Second, "waiting for RT to populated")
+		require.False(t, rt.Satisfied(), "tracker with 2 retries should not be satisfied")
+
+		// observe the sync source for readiness
+		rt.syncsets.Observe(&testSyncSet)
+
+		var retries int
+		if tt.objDataFn == nil {
+			retries = 0
+		} else {
+			retries = tt.objDataFn().retries
+		}
+
+		for i := retries; i > 0; i-- {
+			require.False(t, rt.data.Satisfied(), "data tracker should not be satisfied")
+			require.False(t, rt.Satisfied(), fmt.Sprintf("tracker with %d retries should not be satisfied", i))
+			rt.TryCancelData(podGVK)
+		}
+
```

Comment:
Should assert in between that `rt.Satisfied() == false`.  Otherwise you could have an off-by-one with TryCancelData()